pkgs/development/libraries/science/math/tensorflow-lite/default.nix

```diff
@@ -1,203 +1,119 @@
-  };
-
-  fp16-src = fetchFromGitHub {
-    owner = "Maratyszcza";
-    repo = "FP16";
-    rev = "4dfe081cf6bcd15db339cf2680b9281b8451eeb3";
-    sha256 = "06a8dfl3a29r93nxpp6hpywsajz5d555n3sqd3i6krybb6swnvh7";
-  };
-
-  ruy-src = fetchFromGitHub {
-    owner = "google";
-    repo = "ruy";
-    rev = "23633b37099b614a2f836ef012cafc8087fdb98c";
-    sha256 = "14k9hz6ss8qy8nsajk6lrq25f6qxrldxky31ijw0dpqnfnnswrx4";
-  };
-
-  cpuinfo-src = fetchFromGitHub {
-    owner = "pytorch";
-    repo = "cpuinfo";
-    rev = "5916273f79a21551890fd3d56fc5375a78d1598d";
-    sha256 = "0q6760xdxsg18acdv8vq3yrq7ksr7wsm8zbyan01zf2khnb6fw4x";
+  bazelHostConfigName = {
+    aarch64-linux = "elinux_aarch64";
   };
+  bazelDepsSha256ByHost =
+    bazelDepsSha256ByBuildAndHost.${buildPlatform.system} or
+      (throw "unsupported build system ${buildPlatform.system}");
+  bazelDepsSha256 = bazelDepsSha256ByHost.${hostPlatform.system} or
+      (throw "unsupported host system ${hostPlatform.system} with build system ${buildPlatform.system}");
 in
-stdenv.mkDerivation rec {
-  pname = "tensorflow-lite";
-  version = "2.5.0";
+buildBazelPackage rec {
+  name = "tensorflow-lite";
+  version = "2.13.0";
 
   src = fetchFromGitHub {
     owner = "tensorflow";
     repo = "tensorflow";
     rev = "v${version}";
-    sha256 = "1jdw2i1rq06zqd6aabh7bbm0avsg4pygnfmd7gviv0blhih9054l";
+    sha256 = "sha256-Rq5pAVmxlWBVnph20fkAwbfy+iuBNlfFy14poDPd5h0=";
   };
 
-  patches = [
-    # TODO: remove on the next version bump
-    (fetchpatch {
-      name = "include-schema-conversion-utils-source.patch";
-      url = "https://github.com/tensorflow/tensorflow/commit/f3c4f4733692150fd6174f2cd16438cfaba2e5ab.patch";
-      sha256 = "0zx4hbz679kn79f30159rl1mq74dg45cvaawii0cyv48z472yy4k";
-    })
-    # TODO: remove on the next version bump
-    (fetchpatch {
-      name = "cxxstandard-var.patch";
-      url = "https://github.com/tensorflow/tensorflow/commit/9b128ae4200e10b4752f903492d1e7d11957ed5c.patch";
-      sha256 = "1q0izdwdji5fbyqll6k4dmkzfykyvvz5cvc6hysdj285nkn2wy6h";
-    })
+  bazel = buildPackages.bazel_5;
+
+  nativeBuildInputs = [ pythonEnv buildPackages.perl ];
+
+  bazelTargets = [
+    "//tensorflow/lite:libtensorflowlite.so"
+    "//tensorflow/lite/c:tensorflowlite_c"
+    "//tensorflow/lite/tools/benchmark:benchmark_model"
+    "//tensorflow/lite/tools/benchmark:benchmark_model_performance_options"
   ];
 
-  buildInputs = [ zlib flatbuffers ];
+  bazelFlags = [
+    "--config=opt"
+  ] ++ (if hostPlatform.system != buildPlatform.system
+  then [
+    "--config=${bazelHostConfigName.${hostPlatform.system}}"
+  ]
+    else []
+  );
 
-  dontConfigure = true;
+  bazelBuildFlags = [ "--cxxopt=--std=c++17" ];
 
-  postPatch = ''
-    substituteInPlace ./tensorflow/lite/tools/make/Makefile \
-      --replace /bin/bash ${bash}/bin/bash \
-      --replace /bin/sh ${bash}/bin/sh
-  '';
+  buildAttrs = {
+
+    preBuild = ''
+      patchShebangs .
```

Comment:
Can we avoid scanning all files for shebangs?

Reply:
Yes. See: 96e6dcafd14e0c26fa51b166914040e27269a7e0